unittests/block_buffer_fetch_tests.cpp

```diff
@@ -0,0 +1,138 @@
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-41-60.index" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-61-80.log" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-61-80.index" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-81-100.log" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-81-100.index" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-101-120.log" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-101-120.index" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-121-140.log" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks-121-140.index" ));
+
+            //fetch a block that is in a catalog file
+            fetch_block_buffer(chain, 60, true);
+            fetch_block_buffer(chain, 60, false);
+            //fetch a block that is in the blocks.log file
+            fetch_block_buffer(chain, 145, true);
+            fetch_block_buffer(chain, 145, false);
+            //fetch a block that doesn't exist locally
+            fetch_block_buffer(chain, 200, true);
+            fetch_block_buffer(chain, 200, false);
+        }FC_LOG_AND_RETHROW()
+
+    }
+
+    struct blocklog_version_set {
+        blocklog_version_set(uint32_t ver) { block_log::set_version(ver); };
+        ~blocklog_version_set() { block_log::set_version(block_log::max_supported_version); };
+    };
+
+    BOOST_AUTO_TEST_CASE(fetch_signed_block_v0_test) {
+        try{
+            namespace bfs = boost::filesystem;
+            fc::temp_directory temp_dir;
+
+            // set block log version less than block_log::max_supported_version
+            blocklog_version_set set_version(1);
+
+            TESTER chain(
+                    temp_dir,
+                    [](controller::config& config) {
+                        config.blog.archive_dir        = "archive";
+                        config.blog.stride             = 20;
+                        config.blog.max_retained_files = 5;
+                    },
+                    true);
+
+            //produce less than stride size (20) blocks to avoid block-log-split that sets log version as block_log::max_supported_version
+            chain.produce_blocks(10);
+
+            auto blocks_dir = chain.get_config().blog.log_dir;
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks.log" ));
+            BOOST_CHECK(bfs::exists( blocks_dir / "blocks.index" ));
+
+            //fetch a block that is in blocks.log file
+            fetch_block_buffer(chain, 5, true);
+            fetch_block_buffer(chain, 5, false);
+            //fetch the head block
+            fetch_block_buffer(chain, 10, true);
+            fetch_block_buffer(chain, 10, false);
+            //fetch a block that doesn't exist locally
+            fetch_block_buffer(chain, 20, true);
+            fetch_block_buffer(chain, 20, false);
+
+        }FC_LOG_AND_RETHROW()
+    }
+
```

Comment:
Add some multi-threaded tests. Write/read and get blocks from block_log at the same time. Note boost test is not thread safe so don't use any BOOST_CHECK in threads.

Reply:
Assertion on  https://github.com/EOSIO/eos/blob/epe-437-retrieve-blocks-during-sync/libraries/chain/controller.cpp#L1324 was failed when I tried making write and read blocks log at the same time. Investigating...

Reply:
Fixed.